VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -122,39 +94,45 @@ public void updateDrawState(TextPaint ds) {
                     authorSpan.setSpan(new StyleSpan(Typeface.BOLD), 0, authorSpan.length(),
                             Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
                 }
+                joinableStrings.add(ResourceUtil.getFormattedString(context.getResources(),
+                        R.string.author_attribution_prefix, "author", authorSpan));
+            }
 
-                @StringRes int finalStringRes;
-                HashMap<String, CharSequence> valuesMap = new HashMap<>();
-                valuesMap.put("time", formattedTime);
-                valuesMap.put("author", authorSpan);
-                if (authorLabel != null) {
-                    finalStringRes = authorAttributionLabel.getStringRes();
-                    valuesMap.put("author_label", authorLabel);
-                } else {
-                    finalStringRes = authorAttributionLabel.getNoLabelStringRes();
-                }
-                CharSequence formattedText = trim(ResourceUtil.getFormattedString(
-                        context.getResources(), finalStringRes, valuesMap));
-                // If time is not available, then reduce the whitespaces
-                // surrounding it's placeholder in the template to one.
-                if (TextUtils.isEmpty(formattedText) || TextUtils.isEmpty(authorLabel)) {
-                    formattedText = removeDuplicateWhitespace(formattedText);
-                }
-                text = formattedText;
+            final String authorLabel = authorData.getAuthorLabel();
+            if (!TextUtils.isEmpty(authorLabel)) {
+                joinableStrings.add(ResourceUtil.getFormattedString(context.getResources(),
+                        R.string.text_in_parenthesis, "text", authorLabel));
             }
-        }
 
-        textView.setText(text);
+            int joinableStringsSize = joinableStrings.size();
+            if (joinableStringsSize == 0 || (isEndorsed && joinableStringsSize == 1)) {
+                text = null;
+            } else {
+                text = join(joinableStrings);
+            }
+        }
         if (TextUtils.isEmpty(text)) {
             textView.setVisibility(View.GONE);
         } else {
+            textView.setText(text);
             // Allow ClickableSpan to trigger clicks
             textView.setMovementMethod(new LinkMovementMethod());
         }
     }
 
-    private static CharSequence getRelativeTimeSpanString(@NonNull Context context, long nowMs,
-                                                          long timeMs) {
+    private static CharSequence join(List<CharSequence> list) {
```

Comment:
Well, you could declare a `CharSequence` type parameter, or keep the parameter the same and for each section check if it doesn't implement the `CharSequence` interface then convert it to a `String`. It would probably be a useful utility either way.